### PR TITLE
Unify test method's coding style

### LINKED ISSUE #529: Unify Test Coding Style
At the moment tests have very different coding styles. Some use a `@test` annotation, some use a `test`method prefix, Some use `snake-case`, some use `pascal-`or `camel-case` and mixtures of all those variants. 
This severly limits the readabilty of the tests, so they signatures of the test methods should be unified. 
As the specific style is a matter of taste and/or habit, the best way to find a unified style is to look for what style is used the most at the moment and to adjust the other tests acoordingly. As both these tasks can be tediuous and error prune, when done manually, it's best to use an automated approach by writing scripts to achieve said goals.

- Investigate the most common coding style in tests
- Adjust tests to follow the most common coding style

### PR DESCRIPTION
**Pull request by sonro**
Ensures every PHPUnit test method follows the same style:
- snake_case
- `test_` prefix
- no `@test` entry in docblock
- `void` return type

All other methods in the test classes remain in camelCase. These include:
- `setUp`/`tearDown` methods
- data providers
- protected/private methods
- external functions

---
Some tests were not running due to having a docblock with a missing `*`. For example:
```
/*
 * @test
 */
 ```

Others had typos that were easily visible once changed to snake_case.

---
Closes #529

### CODE REVIEW

File: tests/API/Unit/Baggage/BaggageTest.php
Comment (brettmc) [inline tests/API/Unit/Baggage/BaggageTest.php:84]: half-changed?

**Review by brettmc — APPROVED** [review]
Welcome aboard and thanks for your contribution :)
Changes look good to me, just one comment to look at.

File: tests/SDK/Unit/EnvironmentVariablesTraitTest.php
Comment (brettmc) [inline tests/SDK/Unit/EnvironmentVariablesTraitTest.php:101]: this needs a test_ prefix since the @test annotation is removed

**Review by tidal — APPROVED** [review]

**Comment by tidal** [comment]
@sonro 
Thanks for your contribution.
FYI: I have changed the last line of your PR description, so it automatically closes the respective issue once merged.

**Comment by tidal** [comment]
@brettmc @bobstrecansky 
Please do not merge yet. This is PR is creating a flaky code-coverage report again, while there should be no change at all.
I want to investigate the issue and hopefully fix it. I guess we can adjust [codecov.yml](https://docs.codecov.com/docs/codecovyml-reference) to prevent this from happening.

**Comment by sonro** [comment]
@tidal
Code coverage via `make test` appears consistant with before my changes.

File: tests/SDK/Unit/EnvironmentVariablesTraitTest.php
Comment (sonro) [reply tests/SDK/Unit/EnvironmentVariablesTraitTest.php]: Thank you! This file needed some manual editing as some of the tests had bad docblocks.
Fixed

File: tests/API/Unit/Baggage/BaggageTest.php
Comment (sonro) [reply tests/API/Unit/Baggage/BaggageTest.php]: Weird! Well spotted.
Fixed

**Comment by tidal** [comment]
@sonro 
Yes, that's the problem. We had cases before where codecov reported a drop in code-coverage, while there has not been any change to code or tests at all. This is probably caused by merging the peoprts of the different PHP versions.

Anyway I will merge the PR and postpone investigating this issue.

**Pull request merged**
